model/fv_regional_bc.F90

```diff
@@ -6636,21 +6637,31 @@ subroutine get_data_source(source,regional)
 !
 ! This routine extracts the data source information if it is present in the datafile.
 !
-      character (len = 80) :: source
-      integer              :: ncids,sourceLength
+      character (len=80) :: source      
       logical :: lstatus,regional
 !
 ! Use the fms call here so we can actually get the return code value.
+! The term 'source' is specified by 'chgres_cube'
 !
       if (regional) then
        lstatus = get_global_att_value('INPUT/gfs_data.nc',"source", source)
       else
        lstatus = get_global_att_value('INPUT/gfs_data.tile1.nc',"source", source)
       endif
+      if (mpp_pe()==0) write(*,*) 'INPUT gfs_data source string=',source
       if (.not. lstatus) then
        if (mpp_pe() == 0) write(0,*) 'INPUT source not found ',lstatus,' set source=No Source Attribute'
        source='No Source Attribute'
       endif
+
+! data source string to global string --------
+      if ( trim(source)=='FV3GFS GAUSSIAN NEMSIO FILE' .or.        &
+           trim(source)=='FV3GFS GAUSSIAN NETCDF FILE' .or.        &
+           trim(source)=='FV3GFS GRIB2 FILE'                ) then
+         source = 'FV3GFS NEMSIO/NETCDF/GRIB2'
```

Comment:
Do not assign literal string (`'FV3GFS NEMSIO/NETCDF/GRIB2'`), use `data_source_fv3gfs` parameter.

Reply:
@DusanJovic-NOAA, as mentioned above, I've changed the literal string to the logical flag 'data_source_fv3gfs' to make it easy to maintain multiple files.